app/services/attribute_asserter.rb

```diff
@@ -214,6 +217,13 @@ def add_all_emails(attrs)
     }
   end
 
+  def last_email_from_sp(principal)
+    return nil unless IdentityConfig.store.feature_select_email_to_share_enabled
+    identity = principal.active_identity_for(service_provider)
+    email_id = identity&.email_address_id
+    principal.confirmed_email_addresses.find_by(id: email_id)&.email if identity.email_address_id
```

Comment:
Nit: We're inconsistently using the variable we assigned. Also, we're inconsistently using `nil`-safe operator on `identity`. Will `identity` exist always or not?

Either:

```suggestion
    principal.confirmed_email_addresses.find_by(id: identity.email_address_id)&.email if identity.email_address_id
```

Or:

```suggestion
    email_id = identity.email_address_id
    principal.confirmed_email_addresses.find_by(id: email_id)&.email if email_id
```

Or (if not `nil`-safe):

```suggestion
    email_id = identity&.email_address_id
    principal.confirmed_email_addresses.find_by(id: email_id)&.email if email_id
```